flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalJoin.java

```diff
@@ -276,12 +278,21 @@ private TwoInputStreamOperator<RowData, RowData, RowData> createJoinOperator(
                     isLeftOuterJoin);
         } else {
             if (isTemporalFunctionJoin) {
-                return new TemporalProcessTimeJoinOperator(
-                        InternalTypeInfo.of(rightInputType),
-                        generatedJoinCondition,
-                        minRetentionTime,
-                        maxRetentionTime,
-                        isLeftOuterJoin);
+                if (config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_STATE_ENABLED)) {
```

Comment:
From the other PR, As a heads up, changes like this should update consumedOptions for the JSON serialization.

(Sorry for splitting the conversation across PRs.)

Reply:
The settings related to this configuration have been updated in https://github.com/apache/flink/pull/25320.

I don't quite understand the meaning of what you mean by 'update consumed Options for the JSON serialization'. I would greatly appreciate it if you could give me some more tips.

Reply:
Okay, thanks. I have added the corresponding explanation for consumedOptions.